### PR TITLE
feat: add add how to contribute docs

### PR DESCRIPTION
**Pull request by chouandy**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,62 @@
...
+
+1. **Create a New Logic:**
+  - Add a new folder under `src/logics` named after the protocol (protocolId), using Hyphenation format.
+  - Create a new logic file under `src/logics/{protocolId}`, named with the action name (logicId). Prefix it with `logic.` and use Hyphenation format. The file name should be `logic.{logicId}.ts`.
+
+2. **Implement the Logic:**
+  - Implement the Logic class in the logic file using PascalCase. Append `Logic` to the action name.
+  - Extend the `core.Logic` class.
+  - Implement the following interfaces based on the requirements:
+    - `core.LogicTokenListInterface`: Support for tokens list function.
```
Comment (Jeff-CCH) [inline README.md:26]:
```suggestion
    - `core.LogicTokenListInterface`: Support for tokens list functions.
```

File: README.md [hunk 2]
```diff
@@ -6,6 +6,62 @@
...
+    - `core.LogicOracleInterface`: Quotation function.
```
Comment (Jeff-CCH) [inline README.md:27]:
```suggestion
    - `core.LogicOracleInterface`: Quotation functions.
```

File: README.md [hunk 3]
```diff
@@ -6,6 +6,62 @@
...
+    - `core.LogicBuilderInterface`: Txdata encode function for interacting with contracts.
```
Comment (Jeff-CCH) [inline README.md:28]:
```suggestion
    - `core.LogicBuilderInterface`: Txdata encode functions for interacting with contracts.
```

File: README.md [hunk 4]
```diff
@@ -6,6 +6,62 @@
...
+  - More: [GO](src/logics/)
+
+4. **Unit Testing:**
+  - Write tests that won't change based on block number increments.
+  - Test files are in the same path as the logic files, named `logic.{logicId}.test.ts`.
+  - Write tests for functions like `getTokenList()` and `build()` as needed.
+
+5. **Integration Testing:**
+  - Write tests that interact with the Router contract in real time.
+  - Test files are in `test/logics/{protocolId}/{logicId}.test.ts`.
```
Comment (Jeff-CCH) [inline README.md:52]:
```suggestion
  - Test files are in `test/logics/{protocolId}/`, named `{logicId}.test.ts`.
```

File: README.md [hunk 5]
```diff
@@ -6,6 +6,62 @@
...
+  - Utilize Logic's quotation function to generate Logic Data for contract interaction.
```
Comment (Jeff-CCH) [inline README.md:53]:
```suggestion
  - Utilize Logic's quotation functions to generate Logic Data for contract interactions.
```

File: PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# Pull Request Template
+
+---
+## Protocol Information
+
+- **Name:**
+- **Twitter Link:**
+- **Website Link:**
+- **Github:**
```
Comment (Jeff-CCH) [inline PULL_REQUEST_TEMPLATE.md:9]: Is `Github` provided in an account or a URL?

File: PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+- **Developer Docs:**
```
Comment (Jeff-CCH) [inline PULL_REQUEST_TEMPLATE.md:10]: Is this provided in a URL?

File: README.md [hunk 8]
```diff
@@ -6,6 +6,62 @@
...
+
+3. **Reference Examples:**
+  - Explore reference examples for different categories:
+    - Lending: [Aave V3](src/logics/aave-v2/), [Compound V3](src/logics/comopound-v3/)
+    - Swap: [Uniswap V3](src/logics/uniswap-v3/), [ParaSwap V5](src/logics/paraswap-v5/)
+    - FlashLoan: [Aave V3](src/logics/aave-v3/), [Balancer V2](src/logics/balancer-v2/)
+  - More: [GO](src/logics/)
+
+4. **Unit Testing:**
+  - Write tests that won't change based on block number increments.
```
Comment (Jeff-CCH) [inline README.md:46]:
```suggestion
  - Write tests that won't fail based on block number increments.
```
Is this closer to the intended meaning?

File: PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+- **Are you a member of the protocol?** [Yes/No]
+
+## Checklist:
+
+- [ ] Added export `./{protocolId}` in `src/logics/index.ts`
+- [ ] Exported all Logics in `src/logics/{protocolId}/index.ts`
+- [ ] Ensured that all Logics have unit tests
```
Comment (Jeff-CCH) [inline PULL_REQUEST_TEMPLATE.md:17]:
```suggestion
- [ ] Ensured that all Logics have passed unit tests
```

File: PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+- [ ] Ensured that all Logics have integration tests
```
Comment (Jeff-CCH) [inline PULL_REQUEST_TEMPLATE.md:18]:
```suggestion
- [ ] Ensured that all Logics have passed integration tests
```

File: README.md (hunk 1 above)
Comment (chouandy) [reply README.md:26]: If there's only one function, should it be named in the plural form?

File: README.md (hunk 2 above)
Comment (chouandy) [reply README.md:27]: If there's only one function, should it be named in the plural form?

File: README.md (hunk 3 above)
Comment (chouandy) [reply README.md:28]: If there's only one function, should it be named in the plural form?

File: README.md (hunk 4 above)
Comment (chouandy) [reply README.md:52]: Why?

File: PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (chouandy) [reply PULL_REQUEST_TEMPLATE.md:9]: It seems "GitHub" isn't very intuitive for you. Should we change it to "GitHub Link"?
cc @zodahu

File: PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (chouandy) [reply PULL_REQUEST_TEMPLATE.md:10]: It seems "Developer Docs" isn't very intuitive for you. Should we change it to "Developer Docs Link"?
cc @zodahu

File: README.md (hunk 8 above)
Comment (chouandy) [reply README.md:46]: OK

File: PULL_REQUEST_TEMPLATE.md (hunk 9 above)
Comment (chouandy) [reply PULL_REQUEST_TEMPLATE.md:17]: This check is intended to emphasize the importance of writing tests. It's not about the person submitting the PR checking whether all unit tests are passing.

File: PULL_REQUEST_TEMPLATE.md (hunk 10 above)
Comment (chouandy) [reply PULL_REQUEST_TEMPLATE.md:18]: This check is intended to emphasize the importance of writing tests. It's not about the person submitting the PR checking whether all unit tests are passing.

File: README.md [hunk 11]
```diff
@@ -6,6 +6,62 @@
 
 An SDK that build protocol logics for Protocolink
 
+## How to contribute?
+
+### Initial Setup
+
+1. Fork this repository.
+2. Set up your development environment and install the required dependencies.
```
[the remark this answers is not in the document]
Comment (chouandy) [reply README.md:14]: Could you provide more details about what needs to be added? The only thing that comes to mind initially is after forking the repository, just need to run `yarn`.

File: README.md (hunk 4 above)
Comment (chouandy) [reply README.md:52]: OK

File: README.md (hunk 5 above)
Comment (chouandy) [reply README.md:53]: If there's only one function, should it be named in the plural form?

File: README.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (zodahu) [reply README.md:14]: There is `## Install` at the end of the README.

[the remark this answers is not in the document]
Comment (zodahu) [reply README.md:14]: Oh, the '## Install' section is intended for the client repository. I think adding `yarn` to the CLI section would make more sense.

[the remark this answers is not in the document]
Comment (chouandy) [reply README.md:14]: Updated

File: README.md (hunk 1 above)
Comment (chouandy) [reply README.md]: Updated

File: README.md (hunk 2 above)
Comment (chouandy) [reply README.md]: Updated

File: README.md (hunk 3 above)
Comment (chouandy) [reply README.md]: Updated

File: README.md (hunk 4 above)
Comment (chouandy) [reply README.md]: Updated

File: README.md (hunk 8 above)
Comment (chouandy) [reply README.md]: Updated

File: PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (chouandy) [reply PULL_REQUEST_TEMPLATE.md:10]: @zodahu, what are your thoughts on this?

File: PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (chouandy) [reply PULL_REQUEST_TEMPLATE.md:9]: Updated

File: README.md (hunk 5 above)
Comment (chouandy) [reply README.md]: Updated

File: PULL_REQUEST_TEMPLATE.md (hunk 7 above)
Comment (Jeff-CCH) [reply PULL_REQUEST_TEMPLATE.md:10]: I think keeping it the same is fine from the same rule for Twitter and Website.

**Review by Jeff-CCH — APPROVED** [review]
LGTM

**Pull request merged**